Locations Near User

MigleMic opened the pull request:
- Added locations near user(accessible with button in home page). Now when the user less than 3km away, the locations will appear in specific page. 
- If location is disabled, it will send user to locations page. 
- Removed latitude and longitude from detail page, it now displays the address of the location.
- Used Join requirenment

MigleMic committed:
Locatior near user

MigleMic committed:
Fully working now

MigleMic committed:
Small fix

MigleMic committed:
Removed comments

Waltzerizm reviewed (commented):
I'm not sure why but I couldn't get the "Locations near me" function to work. I tried it on the emulator and my own phone but everytime I would get an empty "Near me" screen without any locations shown.

Waltzerizm commented on Encountify/ViewModels/LocationDetailViewModel.cs:
```diff
@@ -68,40 +69,44 @@ public string Category
-                Longitude = location.Longitude;
                 Latitude = location.Latitude;
+                Longitude = location.Longitude;
+                Position position = new Position(location.Latitude, location.Longitude);
```
Might as well already just use Latitude and Longitude rather than location.Latitude and location.Longitude

Waltzerizm commented on Encountify/Views/HomePage.xaml.cs:
```diff
@@ -16,5 +16,9 @@ public HomePage()
         }
+
+        private async void OnLocationsClicked(object sender, EventArgs e)
+        {
+            await Shell.Current.GoToAsync("//LocationsNearUserPage");
     }
```
missing a bracket

MigleMic committed:
Missing bracket

MigleMic committed:
Small change

MigleMic committed:
Merge branch 'location-near-me' of https://github.com/Mdominykas/EncounterMe-name-subject-to-change into location-near-me

MigleMic committed:
Latitude Longitude fix

Mdominykas commented on Encountify/Services/NearUserCreation.cs:
```diff
@@ -0,0 +1,90 @@
+            
+                foreach (var res in query)
+                {
+                    string end = new string(res.FormattedDistance.Where(c => c != ',' && (c < '0' || c > '9')).ToArray());
```
It would be probably be a good idea to add additional function to DistanceCounter that calculated not formatted distance, cause this way here would be not need to parse the expression.

urtegra committed:
I deserve a prize for this

urtegra committed:
Fix typos

Waltzerizm reviewed (approved):
Seem to be working now, good stuff:-)

MigleMic committed:
Changes

MigleMic commented:
Added some changes that were requested:

- Locations are now sorted by distance as @hickoryeyes asked
- Added function that converts all distance to meters for better comparison. As @Mdominykas mentioned, now there is no need to parse the expression
- Changed the counting of yards (the calculated distance was incorret)
- Now visited locations are not shown in near user page

Waltzerizm commented on Encountify/Services/DistanceCounter.cs:
```diff
@@ -70,5 +72,41 @@ public static void ChangeScale()
+                result *= 1000;
+            }
+
+            if(end.Equals(" m"))
+            {
+                var distance = string.Concat(formattedDistance.Split(remove.ToArray()));
+                Double.TryParse(distance, out result);
+            }
+
+            if(end.Equals(" yd"))
+            {
+                var distance = string.Concat(formattedDistance.Split(remove.ToArray()));
+                Double.TryParse(distance, out result);
+                result /= 1.094;
+            }
+
+            if(end.Equals(" mi"))
```
Could just use "else if" to avoid unnecessary checks

Waltzerizm reviewed (approved):
Works great:-)

MigleMic committed:
Small change

MigleMic committed:
Merge branch 'main' into location-near-me
